src/m365/spo/commands/tenant/tenant-commandset-set.ts

```diff
@@ -120,36 +152,116 @@ class SpoTenantCommandSetSetCommand extends SpoCommand {
         throw 'No app catalog URL found';
       }
 
-      const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, '/lists/TenantWideExtensions');
-      const listItem = await this.getListItemById(logger, appCatalogUrl, listServerRelativeUrl, args.options.id);
+      if (args.options.newClientSideComponentId !== undefined) {
+        const componentManifest = await this.getComponentManifest(appCatalogUrl, args.options.newClientSideComponentId, logger);
+        const clientComponentManifest = JSON.parse(componentManifest.ClientComponentManifest);
+
+        if (clientComponentManifest.extensionType !== "ListViewCommandSet") {
+          throw `The extension type of this component is not of type 'ListViewCommandSet' but of type '${clientComponentManifest.extensionType}'`;
+        }
+
+        const solution = await this.getSolutionFromAppCatalog(appCatalogUrl, componentManifest.SolutionId, logger);
 
-      if (listItem.TenantWideExtensionLocation.indexOf("ClientSideExtension.ListViewCommandSet") === -1) {
-        throw 'The item is not a ListViewCommandSet';
+        if (!solution.ContainsTenantWideExtension) {
+          throw `The solution does not contain an extension that can be deployed to all sites. Make sure that you've entered the correct component Id.`;
+        }
+        else if (!solution.SkipFeatureDeployment) {
+          throw 'The solution has not been deployed to all sites. Make sure to deploy this solution to all sites.';
+        }
       }
 
-      await this.updateTenantWideExtension(appCatalogUrl, args.options, listServerRelativeUrl, logger);
+      const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, '/lists/TenantWideExtensions');
+      const listItemId: number = await this.getListItemId(appCatalogUrl, args.options, listServerRelativeUrl, logger);
+
+      await this.updateTenantWideExtension(logger, appCatalogUrl, args.options, listServerRelativeUrl, listItemId);
     }
     catch (err: any) {
       this.handleRejectedODataJsonPromise(err);
     }
   }
 
-  private async getListItemById(logger: Logger, webUrl: string, listServerRelativeUrl: string, id: string): Promise<any> {
+  private async getComponentManifest(appCatalogUrl: string, clientSideComponentId: string, logger: Logger): Promise<any> {
```

Comment:
I think in my opinion, this method could be refactored as a `util` method since the similar kind of method is being used in `spo applicationcustomizer set` as well.